webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/score/ScoreSerializer.scala

```diff
@@ -0,0 +1,114 @@
+package com.normation.rudder.score
+
+import com.normation.errors._
+import com.normation.errors.IOResult
+import com.normation.rudder.domain.reports.ComplianceSerializable
+import com.normation.zio._
+import zio.Ref
+import zio.ZIO
+import zio.json._
+import zio.json.ast.Json
+import zio.syntax._
+
+object ScoreSerializer {
+
+  implicit val scoreValueEncoder: JsonEncoder[ScoreValue] = JsonEncoder[String].contramap(_.value)
+  implicit val scoreValueDecoder: JsonDecoder[ScoreValue] = JsonDecoder[String].mapOrFail(ScoreValue.fromString)
+
+  implicit val noDetailsScoreEncoder: JsonEncoder[NoDetailsScore] = DeriveJsonEncoder.gen
+  implicit val noDetailsScoreDecoder: JsonDecoder[NoDetailsScore] = DeriveJsonDecoder.gen
+
+  implicit val globalScoreEncoder: JsonEncoder[GlobalScore] = DeriveJsonEncoder.gen
+  implicit val globalScoreDecoder: JsonDecoder[GlobalScore] = DeriveJsonDecoder.gen
+
+  implicit val jsonScoreEncoder: JsonEncoder[JsonScore] = DeriveJsonEncoder.gen
+  implicit val jsonScoreDecoder: JsonDecoder[JsonScore] = DeriveJsonDecoder.gen
+}
+
+trait ScoreSerializerService {
+  def parse(score:  Score[Json]): IOResult[Option[Score[_]]]
+  def toJson(score: Score[_]):    IOResult[Option[Score[Json]]]
```

Comment:
I don't see how it can fail. I understand to potential `None` case if there is no serializer for that score, but if there is one, it should not ever fail (and so just be a `Option[Score[Json]]`

Reply:
I wonder if that can't be simplier. 
It's used in two place: 
- in DB to read/write from a JSON column (so a string), and using the intermediary `Json` means that we do `db -> get String -> parse to Json -> serialize to String -> parse to T` - we could certainly directly use the string, even if that means having a `StringScore` to simplify doobie interaction, 
- in node API, to serialize these info. But here, we should really just need the `JsonEncoder` which we already have (and again avoid the `T -> to String -> parse Json -> to String`)

Reply:
General idea draft in the second commit of that PR: https://github.com/VinceMacBuche/rudder/pull/29 (not sure it floats till the end, but we can in the worst case keep `JsonScore` for `NodeAPI`)

Reply:
I prefer to remove invalid score (that are not json) at the beginning. and it will end up parsed as json (either in api, could be also used to init score if we wanted to recompute a score ) I don't think there is a usage in using a String and not a Json, which is the underlying type

Reply:
We already have everything to manage json from database. and here you have to translate into json at the end which can be already done

Reply:
You can still check for invalid data structure at the begining, it doesn't change that. You certainly can use directly the `JsonEncoder` and `JsonDecoder` in the score repository, I just didn't take the time to implem it. 
Here, each time you use `as` or `toJsonAST`, you really go through a string parsing in the middle. 

And I really believe we should not have one unique JsonScore, it will make the score calculation part less clear.